src/main/java/com/cloudbees/jenkins/plugins/advisor/casc/AdvisorRootConfigurator.java

```diff
@@ -0,0 +1,151 @@
+package com.cloudbees.jenkins.plugins.advisor.casc;
+
+import com.cloudbees.jenkins.plugins.advisor.AdvisorGlobalConfiguration;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.util.FormValidation;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.RootElementConfigurator;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import io.jenkins.plugins.casc.model.Scalar;
+import io.jenkins.plugins.casc.model.Sequence;
+import org.apache.commons.lang.BooleanUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * A root element configurator used for configuring CloudBees Jenkins Advisor through {@link io.jenkins.plugins.casc.ConfigurationAsCode}
+ * https://github.com/jenkinsci/configuration-as-code-plugin
+ */
+@Extension(optional = true)
+@Restricted(NoExternalUse.class)
+public class AdvisorRootConfigurator extends BaseConfigurator<AdvisorGlobalConfiguration> implements RootElementConfigurator<AdvisorGlobalConfiguration> {
+
+    private static final Logger LOG = Logger.getLogger(AdvisorRootConfigurator.class.getName());
+
+    // Ignoring lastBundleResult since it is not configured in the plugin. It only informs about the last bundle generation
+    // Ignoring isValid since it is something auto-acalculated during the configuration
+    private final Collection<String> excludedAttributesInConf = Arrays.asList("lastBundleResult", "valid");
+
+    @NonNull
+    @Override
+    public String getName() {
+        return "advisor";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return getConfiguration().getDisplayName();
+    }
+
+    @Override
+    protected AdvisorGlobalConfiguration instance(Mapping mapping, ConfigurationContext configurationContext) throws ConfiguratorException {
+        // Scalar values
+        final String email = (mapping.get("email") != null ? mapping.getScalarValue("email") : StringUtils.EMPTY);
```

Comment:
looks more complex than is needed (https://stackoverflow.com/a/4095556/4951015), any reason to use empty string over null as well?
```suggestion
        final String email = mapping.get("email") != null ? mapping.getScalarValue("email") : "";
```

Reply:
I like using the `StringUtils.EMPTY` just because of the number of characters. `""` is clear enough always, but I find easier to read/detect that constant in larger lines. Personally I don't like splitting the sentence in several lines unless it is much wider.

Regarding the null vs empty value question, I prefer to use `null` in general, but in this case, I'm mimicking the behaviour from the UI, which send an empty string if the field is not filled in. Doing so, I pretend to prevent the `AdvisorGlobalConfiguration` from handling two different inputs for empty fields in future.